contracts/lazyclaim/ERC721LazyClaim.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+import "@manifoldxyz/creator-core-solidity/contracts/core/IERC721CreatorCore.sol";
+import "@manifoldxyz/libraries-solidity/contracts/access/AdminControl.sol";
+import "@manifoldxyz/creator-core-solidity/contracts/extensions/ICreatorExtensionTokenURI.sol";
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+
+import "./IERC721LazyClaim.sol";
+
+/**
+ * Lazy claim with optional whitelist ERC721 tokens
+ */
+contract ERC721LazyClaim is IERC165, IERC721LazyClaim, ICreatorExtensionTokenURI, ReentrancyGuard {
+  mapping(address => uint) public claimCounts;
+  mapping(address => mapping(uint => Claim)) public claims;
+  mapping(address => mapping(uint => uint)) public mintsPerClaim;
+  mapping(address => mapping(uint => mapping(address => uint))) public mintsPerWallet;
+  mapping(address => mapping(uint => uint)) public tokenClaims;
```

Comment:
Please name better or do a better description.

Reply:
Actually, you should apply my IndexRange trick!
https://github.com/manifoldxyz/creator-core-extensions-solidity/blob/9b9039e6d52237a9a3ec847a807de25af8612165/contracts/manifold/edition/ManifoldERC721Edition.sol#L32

In most cases, since ppl won't be minting in the middle of a claim, it will save a TON of gas.

Reply:
Using IndexRanges since most mints will be sequential.

Keeping the name tokenClaims since that's its primary use, but fully documented the mapping